components/modelregistry/modelregistry.go

```diff
@@ -0,0 +1,222 @@
+// Package modelregistry provides utility functions to config ModelRegistry, an ML Model metadata repository service
+// +groupName=datasciencecluster.opendatahub.io
+package modelregistry
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/go-logr/logr"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	dsciv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/dscinitialization/v1"
+	infrav1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/infrastructure/v1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/components"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/conversion"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/deploy"
+
+	_ "embed"
+)
+
+const DefaultModelRegistryCert = "default-modelregistry-cert"
+
+var (
+	ComponentName                   = "model-registry-operator"
+	DefaultModelRegistriesNamespace = "rhoai-model-registries"
+	Path                            = deploy.DefaultManifestPath + "/" + ComponentName + "/overlays/odh"
+	// we should not apply this label to the namespace, as it triggered namspace deletion during operator uninstall
+	// modelRegistryLabels = cluster.WithLabels(
+	//      labels.ODH.OwnedNamespace, "true",
+	// ).
+)
+
+// Verifies that ModelRegistry implements ComponentInterface.
+var _ components.ComponentInterface = (*ModelRegistry)(nil)
+
+// ModelRegistry struct holds the configuration for the ModelRegistry component.
+// The property `registriesNamespace` is immutable when `managementState` is `Managed`
+
+// +kubebuilder:object:generate=true
+// +kubebuilder:validation:XValidation:rule="(self.managementState != 'Managed') || (oldSelf.registriesNamespace == '') || (oldSelf.managementState != 'Managed')|| (self.registriesNamespace == oldSelf.registriesNamespace)",message="RegistriesNamespace is immutable when model registry is Managed"
+//nolint:lll
+
+type ModelRegistry struct {
+	components.Component `json:""`
+
+	// Namespace for model registries to be installed, configurable only once when model registry is enabled, defaults to "odh-model-registries"
```

Comment:
```suggestion
	// Namespace for model registries to be installed, configurable only once when model registry is enabled, defaults to "rhoai-model-registries"
```

Reply:
i think the reason we keep it as odh-* is to ensure the source code is the same for upstream and downstream. to only config it with different values from csv etc

Reply:
that could be just revert to the previous version instead of all that suggestions.